rtl/eclrtl/rtldynfield.cpp

```diff
@@ -1149,9 +1162,23 @@ class GeneralRecordTranslator : public CInterfaceOf<IDynamicTransform>
             {
                 unsigned matchField = match.matchIdx;
                 const RtlTypeInfo *sourceType = sourceRecInfo.queryType(matchField);
-                size_t sourceOffset = sourceRow.getOffset(matchField);
-                const byte *source = sourceRow.queryRow() + sourceOffset;
-                size_t copySize = sourceRow.getSize(matchField);
+
+                size_t sourceOffset = 0;
+                const byte *source;
+                size_t copySize;
+                if (binarySource)
+                {
+                    const RtlRow &rtlRow = *(const RtlRow *)sourceRow;
+                    sourceOffset = rtlRow.getOffset(matchField);
+                    source = rtlRow.queryRow() + sourceOffset;
+                    copySize = rtlRow.getSize(matchField);
+                }
+                else
+                {
+                    const IDynamicFieldValueFetcher &callbackRowHandler = *(const IDynamicFieldValueFetcher *)sourceRow;
```

Comment:
Could possibly move this to the branch match_typecast|match_dynamic, since I think it is the only place it is used.

Reply:
you're right, I'll move.